src/System.CommandLine/src/System/CommandLine/ArgumentSyntax.cs

```diff
@@ -52,12 +53,7 @@ private void Validate()
                 var helpText = GetHelpText();
                 Console.Error.Write(helpText);
 
-                // TODO: This should use Environment.Exit(0) but this API isn't available yet.
```

Comment:
But you are changing the external behavior of the method. There are people using this already who are expecting this to stop executing when help is requested. If you want to do this, you probably need to introduce an overload or a parameter to not go through this code. And then, new callers can use that. This way you won't break existing users of the API.